packages/portfolio-deploy/src/contract-control.js

```diff
@@ -0,0 +1,308 @@
+ *   privateArgs: Parameters<SF>[1];
+ * }} UpgradeKit
+ */
+
+/**
+ * @param {Zone} zone
+ * @param {{
+ *   agoricNamesAdmin: ERef<NameHubKit['nameAdmin']>,
+ *   board: ERef<Board>,
+ *   zoe: ERef<ZoeService>,
+ *   startUpgradable: ERef<StartUpgradable>,
+ * }} svcs
+ */
+export const prepareContractControl = (zone, svcs) => {
+  return zone.exoClass(
+    'ContractControl',
+    // iface,
+    undefined, // TODO: debug RangeError#13: Expected undefined is same as Interface
+    /**
+     * Install by bundleId and publish to (the board and) agoricNames,
+     * replacing anything that was there before.
+     *
+     * Former installations can be reached from the board.
+     *
+     * @template {ContractStartFunction} SF
+     * @param {{
+     *   name: string;
+     *   storageNode: Remote<StorageNode>;
+     *   kit?: (StartedInstanceKit<SF> & { privateArgs: Parameters<SF>[1] });
+     *   initialPrivateArgs?: Parameters<SF>[1],
+     * }} initial
+     */
+    initial => ({
+      initialPrivateArgs:
+        initial.initialPrivateArgs || initial.kit?.privateArgs,
+      revoked: false,
+      kit: undefined,
+      ...initial,
+    }),
+    {
+      /** @param {string} bundleId */
+      async install(bundleId) {
+        mustMatch(bundleId, iface0.install);
+        const { name, revoked } = this.state;
+        trace(name, 'install', bundleId);
+        !revoked || Fail`revoked`;
+        const { zoe, board, agoricNamesAdmin } = svcs;
+        const installation = await E(zoe).installBundleID(bundleId);
+        const installationAdmin =
+          E(agoricNamesAdmin).lookupAdmin('installation');
+        await E(installationAdmin).update(name, installation);
+        const boardId = await E(board).getId(installation);
+        trace(name, 'installed', { bundleId, installation, boardId });
+        return harden({ boardId, installation });
+      },
+
+      /**
+       * Start the contract; publish the instance to (the board and) agoricNames.
+       *
+       * @throws if already running
+       * @template {ContractStartFunction} SF
+       * @param {object} opts
+       * @param {Installation<SF>} opts.installation
+       * @param {IssuerKeywordRecord} [opts.issuers]
+       * @param {CopyRecord} [opts.privateArgsOverrides]
+       */
+      async start(opts) {
+        mustMatch(opts, iface0.start);
+        const { installation, issuers, privateArgsOverrides } = opts;
+        const { name, storageNode, revoked, initialPrivateArgs } = this.state;
+        !revoked || Fail`revoked`;
+        !this.state.kit || Fail`${name} already started`;
+        const { startUpgradable, board } = svcs;
+        const installationId = await E(board).getId(installation);
+        trace(name, 'startUpgradable', { installation, installationId });
+        const privateArgs = harden({
+          ...initialPrivateArgs,
+          storageNode,
+          ...privateArgsOverrides,
+        });
+        const kit = await E(startUpgradable)({
```

Comment:
As noted elsewhere: we now need to get the `creatorFacet` for the new instance to the control holder.  Just make it retrievable from here?